packages/sdk/src/instant-trade/InstantTradeBuyerTxBuilder.ts

```diff
@@ -0,0 +1,199 @@
+    this.sellerAddress = data.payload && data.payload.address ? data.payload.address : undefined
+    if (!this.sellerAddress) {
+      throw new Error("invalid seller psbt")
+    }
+  }
+
+  private decodePrice() {
+    this.validatePrice((this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0].value - this.postage)
+    this.setPrice((this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0].value - this.postage)
+  }
+
+  private bindRefundableOutput() {
+    this.outputs = [
+      {
+        address: this.address,
+        value: this.utxos.reduce((acc, curr, index) => (acc += [0, 1].includes(index) ? curr.sats : 0), 0)
+      }
+    ]
+  }
+
+  private bindInscriptionOutput() {
+    this.outputs.push({
+      address: this.receiveAddress || this.address,
+      value: this.postage
+    })
+  }
+
+  private mergePSBTs() {
+    const hash = reverseBuffer(this.sellerPSBT.txInputs[0].hash).toString("hex")
+    const index = this.sellerPSBT.txInputs[0].index
+    this.injectableInputs = [
+      {
+        standardInput: {
+          ...this.sellerPSBT.data.inputs[0],
+          hash,
+          index
+        },
+        txInput: (this.sellerPSBT.data.globalMap.unsignedTx as any).tx.ins[0],
+        sats: this.sellerPSBT.data.inputs[0].witnessUtxo?.value,
+        injectionIndex: INSTANT_BUY_SELLER_INPUT_INDEX
+      }
+    ] as unknown as InjectableInput[]
+
+    //outputs
+    this.injectableOutputs = [
+      {
+        standardOutput: this.sellerPSBT.data.outputs[0],
+        txOutput: (this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0],
+        sats: (this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0].value,
+        injectionIndex: INSTANT_BUY_SELLER_INPUT_INDEX
+      }
+    ] as InjectableOutput[]
+  }
+
+  private async findUTXOs() {
+    const utxos = (
+      await OrditApi.fetchUnspentUTXOs({
+        address: this.address,
+        network: this.network,
+        sort: "asc" // sort by ascending order to use low amount utxos as refundable utxos
+      })
+    ).spendableUTXOs.filter((utxo) => utxo.sats >= MINIMUM_AMOUNT_IN_SATS)
+
+    // 3 = 2 refundables + (at least) 1 to cover for purchase
+    if (utxos.length < 3) {
+      throw new Error("No suitable UTXOs found")
+    }
+
+    // bind minimum utxos. PSBTBuilder will add more if needed
+    return utxos.slice(0, 3)
+  }
+
+  async isEligible() {
+    const [utxos] = await Promise.all([this.findUTXOs(), this.verifyAndFindInscriptionUTXO(this.sellerAddress)])
+    const sortedUTXOs = utxos.sort((a, b) => a.sats - b.sats)
+    const [refundableUTXOOne, refundableUTXOTwo, ...restUTXOs] = sortedUTXOs
+    const refundables = [refundableUTXOOne, refundableUTXOTwo].reduce((acc, curr) => (acc += curr.sats), 0)
+    const spendables = restUTXOs.reduce((acc, curr) => (acc += curr.sats), 0)
+    const eligible = refundables >= MINIMUM_AMOUNT_IN_SATS * 2 && spendables >= MINIMUM_AMOUNT_IN_SATS
```

Comment:
not sure why this logic is needed, because if `sortedUTXOs` is already in order, we can just check the index 0 if the sats >= `MINIMUM_AMOUNT_IN_SATS`. If true, then it is "eligible" if not then it will be not "eligible"